eds_scikit/utils/framework.py

```diff
@@ -60,21 +65,142 @@ def pandas(obj: DataObject) -> DataObject:
 
 
 def koalas(obj: DataObject) -> DataObject:
-    if get_framework(obj) is _koalas:
+    if get_framework(obj) is ks:
         return obj
     try:
         return obj.to_koalas()
     except AttributeError:
         pass
 
     # will raise ValueError if impossible
-    return _koalas.from_pandas(obj)
-
-
-def add_unique_id(obj: DataFrame, col_name: str = "id") -> DataFrame:
-    fw = get_framework(obj)
-    if fw == _pandas:
-        obj[col_name] = range(len(obj))
-        return obj
-    else:
-        return obj.koalas.attach_id_column(id_type="distributed", column=col_name)
+    return ks.from_pandas(obj)
+
+
+class BackendDispatcher:
+    """Dispatcher between pandas, koalas and custom method."""
+
+    def get_backend(self, obj: DataObject) -> Optional[ModuleType]:
+        for backend in VALID_FRAMEWORKS:
+            if obj.__class__.__module__.startswith(backend.__name__):
+                return backend
+        return None
+
+    def is_pandas(self, obj: DataObject) -> bool:
+        return self.get_backend(obj) is pd
+
+    def is_koalas(self, obj: DataObject) -> bool:
+        return self.get_backend(obj) is ks
+
+    def to_pandas(self, obj: DataObject) -> DataObject:
+        if self.get_backend(obj) is pd:
+            return obj
+        try:
+            return obj.to_pandas()
+        except AttributeError:
+            raise ValueError("Could not convert object to pandas.")
+
+    def to_koalas(self, obj: DataObject) -> DataObject:
+        if self.get_backend(obj) is ks:
+            return obj
+        try:
+            return obj.to_koalas()
+        except AttributeError:
+            pass
+        # Will raise ValueError if impossible
+        return ks.from_pandas(obj)
+
+    def __getattr__(self, method):
+        """Any method that doesn't belong directly to `BackendDispatcher` will
+         be picked by __getattr__.
+
+        `__getattr__` returns the self.get_params function, wrapped with
+        the queried `method` as a parameter.
+
+         This way, `get_param` will be able to call `method`, dispatched between
+         the desired backend, with the args and kwargs initially provided.
+        """
+        return partial(self.get_params, method)
+
+    def get_params(self, method, *args, backend=None, **kwargs):
+        """
+        This method should only be called by `__getattr__`.
+
+        `get_params` dispatches the call to a backend (pandas or koalas)
+        chosen with simple heuristics.
+        """
+        if isinstance(backend, str):
+            backend = {
+                "pd": pd,
+                "pandas": pd,
+                "ks": ks,
+                "koalas": ks,
+            }[backend]
+        elif backend is pd or backend is ks:
+            pass
+        # TODO: support Dataframe
```

Comment:
Maybe just as simple as that ? :
```suggestion
        elif self.get_backend(backend) is not None:
            backend = self.get_backend(backend)
```

Reply:
Yes, way better!